journal/src/main/java/io/zeebe/journal/file/SegmentedJournalReader.java

```diff
@@ -188,6 +202,10 @@ private boolean hasNextEntry() {
   }
 
   private void replaceCurrentSegment(final JournalSegment nextSegment) {
+    if (currentSegment.equals(nextSegment)) {
+      return;
+    }
+
```

Comment:
This breaks `seekToFirst`

Reply:
That's somewhat unexpected, but looking at the code I guess it makes sense - we expect replacing the current segment to reset the state. It's a bit unintuitive imo, but I imagine this will get fixed when we refactor the impl. I can remove this for now.

Reply:
Actually this passes locally...uh. Let me double check

Reply:
JournalReaderTest#shouldSeekToFirst breaks locally for me.

Reply:
Yes, I was looking at `SegmentedJournalTest#seekToFirst` :see_no_evil: Like I said, I rebased but didn't notice that we had different places where we tested against the API or the implementation (which I like, I just wasn't aware of them).